### PR TITLE
[simpleini] Add package

### PR DESCRIPTION
**Pull request by Qudix**
Added [simpleini](https://github.com/brofield/simpleini)

### CODE REVIEW

**Comment by Qudix** [comment]
Currently debating how to solve the multi-faceted nature of this library in terms of the third party dependency `ConvertUTF.h`.

**Comment by waruqi** [comment]
I uploaded it. https://github.com/xmake-mirror/convertutf

File: packages/s/simpleini/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    set_kind("library", {headeronly = true})
+    set_homepage("https://github.com/brofield/simpleini")
+    set_description("Cross-platform C++ library providing a simple API to read and write INI-style configuration files.")
+    set_license("MIT")
+
+    set_urls("https://github.com/brofield/simpleini/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/brofield/simpleini.git")
+    add_versions("v4.19", "dc10df3fa363be2c57627d52cbb1b5ddd0689d474bf13908e822c1522df8377e")
+
+    add_deps("convertutf", {optional = true})
```
Comment (waruqi) [inline packages/s/simpleini/xmake.lua:11]: please do not use optional, you should add `add_configs()` to enable/disable it and define macro.

https://github.com/brofield/simpleini/blob/142beb4ecef7b50887ae99e7dfcbc09f779372cd/SimpleIni.h#L3008

File: packages/s/simpleini/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    set_kind("library", {headeronly = true})
+    set_homepage("https://github.com/brofield/simpleini")
+    set_description("Cross-platform C++ library providing a simple API to read and write INI-style configuration files.")
+    set_license("MIT")
+
+    set_urls("https://github.com/brofield/simpleini/archive/refs/tags/$(version).tar.gz",
+             "https://github.com/brofield/simpleini.git")
+    add_versions("v4.19", "dc10df3fa363be2c57627d52cbb1b5ddd0689d474bf13908e822c1522df8377e")
+
+    add_configs("convert", {description = "Unicode converter to use.", default = "", type = "string", values = {"", "none", "generic", "icu", "win32"}})
```
Comment (waruqi) [inline packages/s/simpleini/xmake.lua:11]: remove `default = "", `, just use nil

Comment (waruqi) [reply packages/s/simpleini/xmake.lua:11]:
```
add_configs("convert", {description = "Unicode converter to use.", type = "string", values = {"none", "generic", "icu", "win32"}})
```

File: packages/s/simpleini/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    on_load(function (package)
+        if package:config("convert") == "" then
```
Comment (waruqi) [inline packages/s/simpleini/xmake.lua:14]:
```
if not package:config("convert") then
```

**Pull request merged**